### PR TITLE
[Build] Build hdfs fixture faster and less

### PR DESCRIPTION
**Pull request by breskeby**
The building of the shadowed hdfs2 and hdfs3 fixtures takes quite long time due to being 51 and 80mb in size.
By removing non used dependencies from the shadow jar creation we can speed up this significantly.

Also by tweaking the dependency definitions in the dependent projects we can avoid building the expensive shadowed jars for compile dependencies. this safes us a lot especially for precommit checks

### CODE REVIEW

**Comment by breskeby** [comment]
simple benchmark on my local machine of building those hdfs fixture jars:

Screenshot 2024-12-17 at 00 08 11

File: test/fixtures/hdfs-fixture/build.gradle [hunk 1]
```diff
@@ -50,26 +104,16 @@ def hdfs2Jar = tasks.register("hdfs2jar", ShadowJar) {
...
     exclude(dependency("com.carrotsearch.randomizedtesting:randomizedtesting-runner:.*"))
     exclude(dependency("junit:junit:.*"))
-    exclude(dependency("org.slf4j:slf4j-api:.*"))
-    exclude(dependency("com.google.guava:guava:.*"))
-    exclude(dependency("org.apache.commons:commons-compress:.*"))
-    exclude(dependency("commons-logging:commons-logging:.*"))
-    exclude(dependency("commons-codec:commons-codec:.*"))
-    exclude(dependency("org.apache.httpcomponents:httpclient:.*"))
     exclude(dependency("org.apache.httpcomponents:httpcore:.*"))
+//    exclude(dependency("org.apache.httpcomponents:httpclient:.*"))
```
Comment (mark-vieira) [inline test/fixtures/hdfs-fixture/build.gradle:115]: Can this be removed?

File: test/fixtures/hdfs-fixture/build.gradle [hunk 2]
```diff
@@ -86,6 +130,17 @@ tasks.withType(ShadowJar).configureEach {
   }
 }
 
+tasks.register('showHdfs3Jar', Copy) {
```
Comment (mark-vieira) [inline test/fixtures/hdfs-fixture/build.gradle:133]: I don't see what these tasks are used for? Did you add them just to debug what was in these configurations?

Comment (breskeby) [reply test/fixtures/hdfs-fixture/build.gradle:133]: cleaning up. only used for some debugging

File: test/fixtures/hdfs-fixture/build.gradle (hunk 1 above)
Comment (breskeby) [reply test/fixtures/hdfs-fixture/build.gradle]: removed

File: test/fixtures/hdfs-fixture/build.gradle (hunk 2 above)
Comment (breskeby) [reply test/fixtures/hdfs-fixture/build.gradle]: removed

**Review by mark-vieira — APPROVED** [review]

**Pull request merged**